### ISSUE TITLE
Silent message drop since the last maintenance

### ISSUE THREAD

**Issue by airone506** [matrix-org/matrix-appservice-irc#1724]
There is silent message drop since the **Libera.chat bridge** maintenance & upgrade on May 10th, 2023. Estimated ratio of message dropped is ~20% (based on observations, I might add exact numbers later). The Libera.chat messages are not displayed to the Matrix users in the room.

I did some checks in the past, some of them very diligent. I never observed such a massive drop. Actually, I did observe 100% message delivery even on the very diligent checks.

**Comment by airone506** [comment]
I checked quite a busy Libera.Chat channel bridged to Matrix. Within a period of six days, there was only 84.08% delivery success rate, i.e. almost **16% of messages being lost**.

**Comment by Half-Shot** [comment]
Hiya, do you know which channel(s) you're seeing this on. We deployed a patch last week to hopefully improve the issue we were seeing, so sounds like we missed a thing.

**Comment by airone506** [comment]
The issue applies to any bridged channel I'm in. The figures I posted above are from #python. ##English is another Libera channel with some significant traffic volume, where the issue can be clearly seen. Of course it applies for low traffic volume channels like #libera-matrix and #matrix-irc, where I also definitely saw the issue. Matrix room #plasma:kde.org, bridged to Libera channel #plasma also drops the IRC messages...

**Comment by Cydox** [comment]
Also seeing this on the #fedora (on libera.chat, matrix: #fedora:fedoraproject.org)

A seemingly random sample of messages from people on IRC don't appear on Matrix.

**Comment by airone506** [comment]
I see the issue in every Libera channel where I am connected both via IRC client and Matrix. I did last check today for all channels and I can confirm it really is in every channel I'm in.

It seems only a very few users, who are using their Matrix also to access Libera, is bothered by this issue. My guess is they are not aware of the issue.

**Comment by Kleidukos** [comment]
I can confirm that this is currently affecting the #hackage channel on Libera, if you want to see it by yourself.

**Comment by progval** [comment]
This also happens in channels with `"allowUnconnectedMatrixUsers": true`.

**Comment by alkisg** [comment]
I'm also affected by this, e.g. in #debian, #ubuntu, #debian-next, #ubuntu-server etc (of course all in libera).

**Comment by thecb1** [comment]
I saw messages getting lost in both ways btw.

**Comment by airone506** [comment]
> I saw messages getting lost in both ways btw.

This does not match my experience. Weren't these lost messages from Matrix (invisible to Libera) from the users, who did not perform `!reconnect` after May 10th? This would make perfect sense.

For reference. https://github.com/matrix-org/matrix-appservice-irc/issues/1712

**Comment by thecb1** [comment]
Well, it won't reconnect for me ...

**Comment by i-c-o-n** [comment]
Same in #coreboot, virtually nothing from IRC gets through to Matrix.

**Comment by airone506** [comment]
>... virtually nothing from IRC gets through to Matrix.

Yes, this is the current status of the bridge. Way different from the status when this issue has been opened.

**Comment by ApostolosB** [comment]
It seems to be working now. Is this considered fixed?

**Comment by airone506** [comment]
>...  Is this considered fixed?

Definitely not fixed. The issue still appears as it did when this ticket was opened.

**Comment by progval** [comment]
Indeed, I just had a message dropped in `#swh-team:matrix.org` (bridged to `#swh-team` on Libera) at 08:41:42 UTC, in the Libera->Matrix direction. `"allowUnconnectedMatrixUsers": true` is set in this room.

**Comment by AndrewFerr** [comment]
Has this improved at all?  Some fixes were deployed ~10h ago and should have settled down by now.

**Comment by progval** [comment]
I see another loss at 17:42:49 (2.5h ago) on `#swh-sysadm:matrix.org`.

**Comment by progval** [comment]
Also at 20:21:54 on `#libera:libera.chat`

**Comment by airone506** [comment]
> Has this improved at all? Some fixes were deployed...

Thank you for the fixes deployed. I'm sorry to inform the issue is still there.

I just did some quick check in #python Libera channel in the time frame around 18:10 - 18:40 UTC and I saw multiple messages being (seemingly randomly) dropped.

**Comment by progval** [comment]
another one today at 16:21:02, again in `#swh-team:matrix.org` (and it still has `"allowUnconnectedMatrixUsers": true`). right after the dropped message, a puppet connected to IRC.

**Comment by etameta** [comment]
Three drops over 20 messages (so 15%) in `#linux-it:libera.chat` in the last few hours, at `2023-06-06 15:26:08 +0000`, `2023-06-06 18:42:55 +0000` and `2023-06-06 19:49:59 +0000` .

**Comment by Half-Shot** [comment]
I believe this situation has now improved, but please keep letting me know if it's gotten worse.
